File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.util.*;
public class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int arr[] = new int [9];
        int sum = 0;
        int spy1 = 0; int spy2 = 0;
        for(int i = 0; i < 9; i++){
            arr[i] = sc.nextInt();
            sum += arr[i]; //1~9번쨰 애들 키 다 더해져 있음
        }
        Arrays.sort(arr); //키 순으로(오름차순)

        for(int i = 0; i < arr.length-1; i++){
            for(int j = i + 1; j < arr.length; j++){
                if(sum-arr[i]-arr[j] == 100){
                //sum - i번쨰 난쟁이 - j번쨰 난쟁이로 반복하면서 범인찾기
                    spy1 = i;
                    spy2 = j;
                    break;
                }
            }
        }

        for(int i = 0; i < arr.length; i++){
            if(i == spy1 || i == spy2)
                continue; // 범인들은 건너뛰기
            System.out.println(arr[i]);
        }
    }
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
dwarf_with_spy = []

for _ in range(9):
    dwarf_height = int(input())
    dwarf_with_spy.append(dwarf_height)
total = sum(dwarf_with_spy)

spy1, spy2 = 0, 0
for spy1_idx in range(len(dwarf_with_spy)):
    for spy2_idx in range(len(dwarf_with_spy)):
        if spy1_idx == spy2_idx:
            continue
        temp_sum = dwarf_with_spy[spy1_idx] + dwarf_with_spy[spy2_idx]
        if total - temp_sum == 100:
            spy1 = dwarf_with_spy[spy1_idx]
            spy2 = dwarf_with_spy[spy2_idx]
            break

dwarf_with_spy.sort()
for dwarf in dwarf_with_spy:
    if dwarf == spy1 or dwarf == spy2:
        continue
    print(dwarf)